### PR TITLE
feat(backend/executor): persist activity-status LLM cost to PlatformCostLog

### PR DESCRIPTION
**Pull request by majdyz**
## Why

The post-execution **activity-status generator** ([`activity_status_generator.py`](autogpt_platform/backend/backend/executor/activity_status_generator.py)) runs an LLM call (gpt-4o-mini via the platform's `openai_internal_api_key`) on every completed graph run to produce a 1-3 sentence user-friendly summary + correctness score. It uses `AIStructuredResponseGeneratorBlock` to issue the call.

**Bug:** the block writes its `provider_cost` into a local `NodeExecutionStats` via `merge_stats`, but because the block runs **outside** the executor's `execute_node` loop, neither [`log_system_credential_cost`](autogpt_platform/backend/backend/executor/cost_tracking.py) nor `charge_reconciled_usage` ever fires — the cost is dropped on the floor when the function returns.

This is an **observability gap, not a billing gap**: the platform is paying OpenAI for activity-status generation on every completed graph execution, but those calls don't show up in the admin cost dashboard / per-user attribution.

The [simulator](autogpt_platform/backend/backend/executor/simulator.py) (same shape: platform-paid LLM call on the user's behalf) already routes through `persist_and_record_usage` so its spend is attributed correctly. This PR brings activity-status to parity.

## What

- New `_persist_activity_status_cost(...)` helper that builds a `PlatformCostEntry` and schedules it via `schedule_platform_cost_log` (renamed from the previously-private `_schedule_log` — see refactor commit) so external modules can use it without reaching into private API.
- Call site placed after `structured_block.run()` succeeds, before `return activity_response`. Reads `provider_cost`, `input_token_count`, `output_token_count` off `structured_block.execution_stats`.
- Helper body is wrapped in `try/except Exception` so any cost-log failure (transient DB / scheduling error) is logged but never strips a successful activity-status response from the user — cost-logging is strictly best-effort.
- Early-return guard uses `not cost_usd` so both `provider_cost is None` and `provider_cost == 0.0` (with zero tokens) short-circuit, avoiding empty rows that would dilute dashboard averages.
- Distinguishes `cost_usd`-tracked rows (`tracking_type="cost_usd"`) from tokens-only rows (`tracking_type="tokens"`, `tracking_amount = input + output`) so admins can still filter by request volume when the provider doesn't report a USD cost.
- Deliberately **does not** bill the user's wallet (no `spend_credits`) and **does not** count against the user's copilot rate-limit (no `record_cost_usage`) — activity-status is platform-side overhead, not user-triggered. Matches the simulator's stance for dry-run executions.

## How (tests)

`backend/executor/activity_status_generator_test.py`:
- `test_generate_status_persists_platform_cost`: stubs `execution_stats=NodeExecutionStats(input=120, output=40, provider_cost=0.0042)`, patches `schedule_platform_cost_log`, and asserts the entry carries the right `user_id` / `graph_exec_id` / `graph_id` / `block_name="activity_status_generator"` / `provider="openai"` / `model="gpt-4o-mini"` / `tracking_type="cost_usd"` / `cost_microdollars=4200` / `metadata.source="activity_status_generator"`.
- `test_generate_status_no_cost_no_log`: zero-cost zero-token case must skip the log write.
- `test_generate_status_tokens_only_branch`: provider returns no USD cost but tokens are present (input=200, output=80) — entry is logged with `tracking_type="tokens"`, `tracking_amount=280.0`, `cost_microdollars=None`.
- Updated three existing success-path tests to set `mock_instance.execution_stats = NodeExecutionStats()` so the new short-circuit has concrete numbers to compare against (was a `MagicMock` attribute before).

`poetry run pytest backend/executor/activity_status_generator_test.py` — 17 passed locally.
`poetry run pyright` / `poetry run ruff` — clean on touched files.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: autogpt_platform/backend/backend/executor/activity_status_generator.py [hunk 1]
```diff
@@ -367,6 +385,55 @@ async def generate_activity_status_for_execution(
...
+    if cost_usd is None and input_tokens == 0 and output_tokens == 0:
+        return
+
+    cost_microdollars = usd_to_microdollars(cost_usd) if cost_usd is not None else None
+    if cost_usd is not None:
+        tracking_type = "cost_usd"
+        tracking_amount = float(cost_usd)
+    else:
+        tracking_type = "tokens"
+        tracking_amount = float(input_tokens + output_tokens)
```
Comment (majdyz) [inline autogpt_platform/backend/backend/executor/activity_status_generator.py]: 🤖 🟡 **Nice to Have**: The `tracking_type="tokens"` branch (when `stats.provider_cost is None` but tokens > 0) is implemented at lines 414-416 but has no test. Coverage exists for the `cost_usd`-set path (`test_generate_status_persists_platform_cost`) and the both-empty short-circuit (`test_generate_status_no_cost_no_log`), but the tokens-only branch is untested — so a refactor that flipped the branch order (or accidentally broke the `tracking_type` assignment) would slip through. Worth adding one parametrize case or a third test that stubs `NodeExecutionStats(input_token_count=200, output_token_count=80, provider_cost=None)` and asserts `entry.tracking_type == "tokens"` and `entry.tracking_amount == 280`.

File: autogpt_platform/backend/backend/executor/activity_status_generator.py [hunk 2]
```diff
@@ -358,6 +373,18 @@ async def generate_activity_status_for_execution(
...
+        _persist_activity_status_cost(
+            stats=structured_block.execution_stats,
+            user_id=user_id,
+            graph_exec_id=graph_exec_id,
+            graph_id=graph_id,
+            model_name=model_name,
+            db_client=db_client,
+        )
+
         return activity_response
```
[the remark this answers is not in the document]
Comment (majdyz) [reply autogpt_platform/backend/backend/executor/activity_status_generator.py:379]: 🤖 Fixed in [87c28c7ed](https://github.com/Significant-Gravitas/AutoGPT/commit/87c28c7eddb1c81bd3e2b5bbed6f04d75e30d8cf): wrapped `_persist_activity_status_cost` body in try/except so any failure (scheduling, DB) is logged but never propagates to the broad `try/except Exception` in the caller — a successful activity-status response is no longer discarded by a transient cost-log error.

File: autogpt_platform/backend/backend/executor/activity_status_generator.py [hunk 3]
```diff
@@ -367,6 +394,57 @@ async def generate_activity_status_for_execution(
...
+    Mirrors ``backend.copilot.token_tracking._schedule_cost_log``: the platform
+    pays for this call (system OpenAI key) but the user is not billed and not
+    rate-limited, so we skip ``record_cost_usage`` and only write to
+    ``PlatformCostLog`` for admin attribution.
+    """
+    cost_usd = stats.provider_cost
+    input_tokens = stats.input_token_count or 0
+    output_tokens = stats.output_token_count or 0
+    if cost_usd is None and input_tokens == 0 and output_tokens == 0:
+        return
```
[the remark this answers is not in the document]
Comment (majdyz) [reply autogpt_platform/backend/backend/executor/activity_status_generator.py]: 🤖 Fixed in [87c28c7ed](https://github.com/Significant-Gravitas/AutoGPT/commit/87c28c7eddb1c81bd3e2b5bbed6f04d75e30d8cf): changed the guard to `not cost_usd` so `provider_cost == 0.0` short-circuits alongside `None` instead of writing an empty cost_usd row that dilutes dashboard averages.

File: autogpt_platform/backend/backend/executor/activity_status_generator.py (hunk 1 above)
Comment (majdyz) [reply autogpt_platform/backend/backend/executor/activity_status_generator.py]: 🤖 Fixed in [87c28c7ed](https://github.com/Significant-Gravitas/AutoGPT/commit/87c28c7eddb1c81bd3e2b5bbed6f04d75e30d8cf): added `test_generate_status_tokens_only_branch` covering the `provider_cost=None`, `input_tokens=200`, `output_tokens=80` case — asserts `tracking_type='tokens'`, `tracking_amount=280.0`, `cost_microdollars is None`.
